dask_expr/expr.py

```diff
@@ -1015,6 +1015,23 @@ class Assign(Elemwise):
     def _node_label_args(self):
         return [self.frame, self.key, self.value]
 
+    def _simplify_up(self, parent):
+        if isinstance(parent, Projection):
+            columns = (
+                parent.columns
+                if isinstance(parent.columns, pd.Index)
+                else [parent.columns]
```

Comment:
I'm becoming increasingly concerned about the `list/pd.Index/string` type uncertainty on columns.

If you have any interest in looking into this and making a proposal I think it would be time well spent (I can do it too though).

Reply:
Can do it tomorrow!